src/box/txn_limbo.c

```diff
@@ -1304,6 +1338,134 @@ txn_limbo_update_system_spaces_is_sync_state(struct txn_limbo *limbo,
+	port_c_create(out);
+	/*
+	 * The iterator returns the following transaction statement information:
+	 *  1. An ordinal request number;
+	 *  2. The term during which the transaction occurred.
+	 *  3. The ID of the transaction;
+	 *  4. The ID of the limbo owner or 0 for local rows that are part of a
+	 *  global transaction;
+	 *  5. The LSN of the request.
+	 *  6. The ID of the space to which the request was applied.
+	 *  7. The request type.
+	 *  8. The argument of the request depending on the operation type
+	 *  9. An optional argument for the UPDATE/UPSERT request, may be nil.
+	 *  10. The tuple which was replaced by the operation, may be nil.
+	 *  11. The tuple created by the operation, may be nil.
+	 */
+	port_c_add_number(out, entry_it->req_num++);
+	port_c_add_number(out, (uint32_t)txn_limbo.promote_greatest_term);
+	port_c_add_number(out, (uint32_t)stmt->txn->id);
+	port_c_add_number(out, row->replica_id);
+	port_c_add_number(out, (uint32_t)row->lsn);
+	port_c_add_number(out, space_id(stmt->space));
+	port_c_add_number(out, row->type);
+
+	struct request request;
+	VERIFY(xrow_decode_dml(row, &request,
+			       dml_request_key_map(row->type)) == 0);
+	struct tuple_format *format = box_tuple_format_default();
+	if (request.key != NULL) {
+		struct tuple *tuple =
+			box_tuple_new(format, request.key, request.key_end);
+		if (tuple == NULL)
+			return -1;
+		port_c_add_tuple(out, tuple);
+	}
+	if (request.tuple != NULL) {
+		struct tuple *tuple =
+			box_tuple_new(format, request.tuple, request.tuple_end);
+		if (tuple == NULL)
+			return -1;
+		port_c_add_tuple(out, tuple);
+	}
+	if (request.ops != NULL) {
+		struct tuple *tuple =
+			box_tuple_new(format, request.ops, request.ops_end);
+		if (tuple == NULL)
+			return -1;
+		port_c_add_tuple(out, tuple);
+	}
+	if (request.type != IPROTO_UPDATE && request.type != IPROTO_UPSERT)
+		port_c_add_null(out);
+
+	if (stmt->old_tuple != NULL)
+		port_c_add_tuple(out, stmt->rollback_info.old_tuple);
+	else
+		port_c_add_null(out);
+	if (stmt->new_tuple != NULL)
+		port_c_add_tuple(out, stmt->rollback_info.new_tuple);
+	else
+		port_c_add_null(out);
+
+	entry_it->stmt = stailq_next_entry(stmt, next);
+	return 0;
+}
+
+static void
+txn_limbo_entry_iterator_create(void *base_data, struct port_c_iterator *it)
+{
+	struct txn_limbo_entry_port_c_iterator *entry_it =
+		(struct txn_limbo_entry_port_c_iterator *)it;
+	entry_it->next = txn_limbo_entry_iterator_next;
+	struct txn_limbo_entry_iterator_data *data =
+		(struct txn_limbo_entry_iterator_data *)base_data;
+	entry_it->stmt = data->first_stmt;
+	entry_it->req_num = 1;
+}
+
+static int
+on_split_brain_rollback_run_triggers(struct txn_limbo *limbo, int64_t lsn)
+{
+	struct txn_limbo_entry *last_rollback =
+		txn_limbo_last_rollback_entry(limbo, lsn);
+	if (last_rollback == NULL)
+		return 0;
+	struct txn_limbo_entry *e;
+	rlist_foreach_entry_reverse(e, &limbo->queue, in_queue) {
+		if (!txn_has_flag(e->txn, TXN_WAIT_ACK) ||
```

Comment:
What about the `TXN_WAIT_SYNC` transactions? It seems we could put them into the repair queue as well. Would this cause some inconsistency?